lib/singlepass-backend/src/codegen_x64.rs

```diff
@@ -239,6 +239,25 @@ pub struct X64ExecutionContext {
     msm: ModuleStateMap,
 }
 
+/// On-disk cache format.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct CacheImage {
+    /// Code for the whole module.
+    code: Vec<u8>,
+
+    /// Offsets to the beginnings of each function. (including trampoline, if any)
+    function_pointers: Vec<usize>,
+
+    /// Offsets to the beginnings of each function after trampoline.
```

Comment:
Is this the absolute offset in the file of the function, or is this the number of bytes between the function and the trampoline? Also does this mean that the order in the on-disk file is <function1, trampoline1>, <function2, trampoline2>, ...?

Reply:
The absolute offset.

It is <trampoline1, function1>, <trampoline2, function2>, ... , though the trampolines only exist on AArch64.

Reply:
Thanks! Could you clarify this in the comments? Maybe at the top with:
```
/// On-disk cache format.
/// Offsets are relative to the start of the file.
/// Trampolines are only present on AArch64.
```